src/hotspot/share/oops/klassVtable.cpp

```diff
@@ -1174,6 +1175,7 @@ void klassItable::initialize_itable(GrowableArray<Method*>* supers) {
 void klassItable::check_constraints(GrowableArray<Method*>* supers, TRAPS) {
 
   assert(_size_method_table == supers->length(), "wrong size");
+  HandleMark hm(THREAD);
```

Comment:
Question: Would this make the handles area grow up to `_size_method_table` in the worst case? I thought we want to have `HandleMark`-s closer to actual handle creation/last-use to keep the footprint at bay.

Reply:
I was optimising for runtime performance rather than footprint, but you are right this could grow in a way we don't want. So I reverted this change and made a similar change to `klassVtable::check_constraints`.

Thanks for looking at this @shipilev .